commons/src/main/java/org/infinispan/commons/marshall/ProtoStreamTypeIds.java

```diff
@@ -0,0 +1,80 @@
+package org.infinispan.commons.marshall;
+
+/**
+ * TypeIds used by protostream in place of FQN message/enum names to reduce payload size.
+ * <p>
+ * ONCE SET VALUES IN THIS CLASS MUST NOT BE CHANGED AS IT WILL BREAK BACKWARDS COMPATIBILITY.
+ * <p>
+ * Values must be greater than 1000000, as this is the range reserved for internal infinispan use by the protostream
```

Comment:
1000000 is a value I came up after an intense decision making process that involved coin tossing. It's now the perfect time to review that and see if we want to reserve a lower range maybe. One that is large enough and that is also efficiently encoded as an int32 with protobuf's zigzag encoding. The range 1000000 to 1005000 would encode to 3 bytes, which is 1 byte lower than a full 32 bit int, so I thought we're fine. But do we want to be even cheaper?

Reply:
@anistor As cheap as possible ideally. I was curious why we started at `1000000` and not `0`

Reply:
Reserving something for internal use should also take into account external usages. that's why 0 is not a good start for the reserved range. Maybe we should think about selecting a smaller (round) value that encodes to 2 bytes, but I would not go lower.

Reply:
My thoughts were that now protostream is essential to the core of Infinispan, and will only become more important once the GlobalMarshaller uses it, it makes sense to prioritise Infinispan as the primary user in this case.

An additional bytes per user object is not going to have as great an impact on performance. Whereas, one less byte per Infinispan internal object will have a noticeable impact on marshalling size/performance over time and will impact all users, even those using basic types.

Reply:
True. Then we could reserve 0..65535 for internal ISPN use and leave the rest for user types. I'll update the javadocs to reflect that. And also create a jira so this remains recorded.

Reply:
Great, thanks